### ISSUE TITLE
unexpected LFTri behavior with negative freq argument

### ISSUE THREAD

**Issue by elifieldsteel** [supercollider/supercollider#4118]
Environment
-----------

* Your SuperCollider version: 3.9.3
* Your operating system and version: macOS 10.11.6

Steps to reproduce (for bugs)
-----------------------------

```supercollider
// Please paste SuperCollider code here.
// It really helps if you try to simplify your example as much as possible.
```

s.boot;

x = {LFSaw.kr(0.5).poll;0}.play; //fine
x.free;

x = {LFTri.kr(0.5).poll;0}.play; //fine
x.free;

x = {LFSaw.kr(-0.5).poll;0}.play; //fine
x.free;

x = {LFTri.kr(-0.5).poll;0}.play; //descends to -inf
x.free;

Error message (for bugs)
------------------------

```
// Please paste any error messages here **in their entirety**.
// If this is a SuperCollider error message, include the full stack trace.
// Link to a Gist (https://gist.github.com) if the message is long.
```

Expected Behavior
-----------------

//When given a negative freq argument, LFTri should undergo a polarity inversion.

Current Behavior
----------------

//The output of LFTri descends to negative infinity when given a negative freq argument.

**Comment by telephon** [comment]
Yes, it should distinguish the two cases, like other (but not all) LF-Ugens. 

here is the bit of code of `LFSaw`:

```C
if (freq >= 0.f) {
		LOOP1(inNumSamples,
			ZXP(out) = phase;
			phase += freq;
			if (phase >= 1.f) phase -= 2.f;
		);
	} else {
		LOOP1(inNumSamples,
			ZXP(out) = phase;
			phase += freq;
			if (phase <= -1.f) phase += 2.f;
		);
	}
```

**Comment by mtmccrea** [comment]
> When given a negative freq argument, LFTri should undergo a polarity inversion.

Strictly speaking, this isn't a polarity inversion, phase advances backwards. It just looks inverted because you're seeing the output in time, which always move forward :) In a similar way, a half-cycle initial phase offset will look polarity inverted:

`a = {LFTri.kr(10, 2)}.plot(duration:0.15).plotMode_(\points)`

This raises an issue that's actually more prevalent than just `LFTri`. There are a few UGens which don't wrap initial phase so produce strange results based on both the `iphase` and `freq`:
```
{LFTri.kr(10, 35)}.plot(duration:0.15).plotMode_(\points)  // bug: phase doesn't wrap
{LFTri.kr(-10, 4)}.plot(duration:2).plotMode_(\points)     // bug: neg freq output goes to infinity
{LFSaw.kr(-10, 4)}.plot(duration:0.2).plotMode_(\points)   // bug: out of range phase paired with negative freq
{Impulse.kr(10, 3.8)}.plot(duration:0.03);                // bug phase is out of range for 3 k-periods, so outputs 1 during that time
{Impulse.kr(10, -1)}.plot(duration:0.3)                   // bug: negative phase isn't handle properly
```

There's work to be done here!
